journalapp/test/test_views.py

```diff
@@ -0,0 +1,90 @@
+"""Testing views."""
+# _*_.utf8_*_
+from journalapp.views import list_view, detail_view
+from pyramid.testing import DummyRequest
+from journalapp.models import DBSession, Entry
+
+
+def test_list_view(dbtransaction, new_entry):
+    """Test that list_view returns a Query of Entries."""
+    test_request = DummyRequest()
+
+    response_dict = list_view(test_request)
+    entries = response_dict['entries']
+    assert entries[0] == new_entry
+
+
+def test_list_route(dbtransaction, app, new_entry):
+    """Test if model initialized with correct vals."""
+    response = app.get('/')
+    assert response.status_code == 200
+
+
+def test_detail_view(dbtransaction, new_entry):
+    """Test that list_view returns a Query of Entries."""
+    test_request = DummyRequest()
+    test_request.matchdict = {'detail_id': new_entry.id}
+
+    response_dict = detail_view(test_request)
+    assert response_dict['entry'] == new_entry
+
+
+def test_detail_route(dbtransaction, app, new_entry):
+    """Test if model initialized with correct vals."""
+    new_entry_id = new_entry.id
+    response = app.get('/detail/{}'.format(new_entry_id))
+    assert response.status_code == 200
+
+
+def test_list_detail(dbtransaction, new_entry):
+    """Combine tests of list and detail views."""
+    test_request = DummyRequest()
+
+    list_response_dict = list_view(test_request)
+    entry = list_response_dict['entries'][0]
+
+    test_request.matchdict = {'detail_id': entry.id}
+    entry_response_dict = detail_view(test_request)
+    assert entry_response_dict['entry'] == entry == new_entry
```

Comment:
What does this test prove that isn't already proven by some combination of the test above it?